### ISSUE TITLE
segmentation fault immediately following authentication

### ISSUE THREAD

**Issue by chessai** [dlenski/openconnect#129]
I forked `openconnect`, and am using the `master` branch, here: https://github.com/chessai/openconnect, commit is 411f6bab471982c68ddd456a06e897d5eb71a10d

I followed the build instructions for Debian/Ubuntu exactly (I am on Ubuntu 16.04)

I ran it as follows: `sudo openconnect --protocol=gp vpn.somecompany.com`
After login, it fails immediately with: 

```bash
$ sudo ./openconnect --protocol=gp vpn.somecompany.com
Please enter your username and password
Username: myusername
Password:
POST https://vpn.somecompany.com/ssl-vpn/login.esp
Connected to someip:443
SSL negotiation with vpn.somecompany.com
Connected to HTTPS on vpn.somecompany.com
GlobalProtect login returned authentication-source=DuoProxy
POST https://vpn.somecompany.com/ssl-vpn/getconfig.esp
Tunnel timeout (rekey interval) is 180 minutes.
Idle timeout is 180 minutes.
Segmentation fault (core dumped)
```

A coworker is using an older commit, and was able to tell me which commit it was: abb4ef3bf7023e6b52b17e875c0320a262a7f9a7

I checked out that commit, built it in the same way, and the application succeeded.

**Comment by chessai** [comment]
```txt
sudo ./openconnect --protocol=gp --dump -vvv vpn.somecompany.com
Please enter your username and password
Username: myusername
Password: 
POST https://vpn.somecompany.com/ssl-vpn/login.esp
Attempting to connect to server SOMEIP:443
Connected to SOMEIP:443
SSL negotiation with vpn.somecompany.com
Connected to HTTPS on vpn.somecompany.com
> POST /ssl-vpn/login.esp HTTP/1.1
> Host: vpn.somecompany.com
> User-Agent: PAN GlobalProtect
> X-Pad: 00000000000000000000000000000000000000000000000000000000000
> Content-Type: application/x-www-form-urlencoded
> Content-Length: 197
> 
> jnlpReady=jnlpReady&ok=Login&direct=yes&clientVer=4100&prot=https:&clientos=Windows&os-version=linux-64&server=vpn.somecompany.com&computer=dcartwright3520&user=dcartwright&passwd=mypassword
Got HTTP response: HTTP/1.1 200 OK
Date: Thu, 04 Oct 2018 21:03:46 GMT
Content-Type: application/xml; charset=UTF-8
Content-Length: 629
Connection: keep-alive
ETag: "SOME_ETAG"
Pragma: no-cache
Cache-Control: no-store, no-cache, must-revalidate, post-check=0, pre-check=0
Expires: Thu, 19 Nov 1981 08:52:00 GMT
X-FRAME-OPTIONS: DENY
Set-Cookie: PHPSESSID=SOMESESSID; secure; HttpOnly
Set-Cookie: PHPSESSID=SOMESESSID; secure; HttpOnly
Set-Cookie: PHPSESSID=SOMESESSID; secure; HttpOnly
Set-Cookie: PHPSESSID=SOMESESSID; secure; HttpOnly
Set-Cookie: PHPSESSID=SOMESESSID; secure; HttpOnly
Set-Cookie: PHPSESSID=SOMESESSID; secure; HttpOnly
Set-Cookie: PHPSESSID=SOMESESSID; secure; HttpOnly
Set-Cookie: PHPSESSID=SOMESESSID; secure; HttpOnly
Set-Cookie: PHPSESSID=SOMESESSID; secure; HttpOnly
Set-Cookie: PHPSESSID=SOMESESSID; secure; HttpOnly
Set-Cookie: PHPSESSID=SOMESESSID; secure; HttpOnly
Set-Cookie: PHPSESSID=SOMESESSID; secure; HttpOnly
Set-Cookie: PHPSESSID=SOMESESSID; secure; HttpOnly
Set-Cookie: PHPSESSID=SOMESESSID; secure; HttpOnly
Set-Cookie: PHPSESSID=SOMESESSID; secure; HttpOnly
Set-Cookie: PHPSESSID=SOMESESSID; secure; HttpOnly
Set-Cookie: PHPSESSID=SOMESESSID; secure; HttpOnly
Set-Cookie: PHPSESSID=SOMESESSID; secure; HttpOnly
Set-Cookie: PHPSESSID=SOMESESSID; secure; HttpOnly
Set-Cookie: PHPSESSID=SOMESESSID; secure; HttpOnly
Set-Cookie: PHPSESSID=SOMESESSID; secure; HttpOnly
Set-Cookie: PHPSESSID=SOMESESSID; secure; HttpOnly
Set-Cookie: PHPSESSID=SOMESESSID; secure; HttpOnly
HTTP body length:  (629)
< <?xml version="1.0" encoding="utf-8"?><jnlp><application-desc><argument>(null)</argument><argument>SOMEHASH</argument><argument>SOMEHASH1</argument><argument>NAMEOFNETWORK</argument><argument>dcartwright</argument><argument>DuoProxy</argument><argument>vsys2</argument><argument>(empty_domain)</argument><argument>(null)</argument><argument></argument><argument></argument><argument></argument><argument>tunnel</argument><argument>-1</argument><argument>4100</argument><argument></argument><argument></argument><argument></argument><argument></argument></application-desc></jnlp>
GlobalProtect login returned authentication-source=DuoProxy
POST https://vpn.somecompany.com/ssl-vpn/getconfig.esp
> POST /ssl-vpn/getconfig.esp HTTP/1.1
> Host: vpn.somecompany.com
> User-Agent: PAN GlobalProtect
> Cookie: PHPSESSID=SOMESESSID
> X-Pad: 000000000000000000000000000000000000000000000
> Content-Type: application/x-www-form-urlencoded
> Content-Length: 275
> 
> client-type=1&protocol-version=p1&app-version=3.0.1-10&clientos=Windows&os-version=linux-64&hmac-algo=sha1%2cmd5&enc-algo=aes-128-cbc%2caes-256-cbc&authcookie=SOMESESSID&portal=NAMEOFNETWORK&user=dcartwright&domain=%28empty_domain%29&computer=dcartwright3520
Got HTTP response: HTTP/1.1 200 OK
Date: Thu, 04 Oct 2018 21:03:46 GMT
Content-Type: application/xml; charset=UTF-8
Content-Length: 1741
Connection: keep-alive
ETag: "cf24-1f3-5a3ffe84"
Pragma: no-cache
Cache-Control: no-store, no-cache, must-revalidate, post-check=0, pre-check=0
Expires: Thu, 19 Nov 1981 08:52:00 GMT
X-FRAME-OPTIONS: DENY
HTTP body length:  (1741)
< 
<     <response status="success">
<         <need-tunnel>yes</need-tunnel>
<         <ssl-tunnel-url>/ssl-tunnel-connect.sslvpn</ssl-tunnel-url>
<         <portal>NAMEOFNETWORK</portal>
<         <user>dcartwright</user>
<         <lifetime>2592000</lifetime>
<         <timeout>10800</timeout>
<         <disconnect-on-idle>10800</disconnect-on-idle>
<         <bw-c2s>1000</bw-c2s>
<         <bw-s2c>1000</bw-s2c>
<         <gw-address>SOMEIP</gw-address>
<         <ip-address>SOMEIP1</ip-address>
<         <netmask>255.255.255.255</netmask>
<         <dns>
<             <member>SOMEIP2</member>
<             <member>SOMEIP3</member>
<         </dns> 
<         <wins>
<         </wins> 
<         <dns-suffix>
<         </dns-suffix> 
<         <default-gateway>SOMEIP1</default-gateway>
<         <mtu>0</mtu>
<         <no-direct-access-to-local-network>no</no-direct-access-to-local-network>
<         <access-routes>
<             <member>SOMEIPRANGE</member>
<             <member>SOMEIPRANGE1</member>
<             <member>SOMEIPRANGE2</member>
<             <member>SOMEIPRANGE3</member>
<             <member>SOMEIPRANGE4</member>
<             <member>SOMEIP2/32</member>
<             <member>SOMEIP3/32</member>
<         </access-routes> 
<         <exclude-access-routes>
<         </exclude-access-routes> 
<         <ipsec>
<             <udp-port>4501</udp-port>
<             <ipsec-mode>esp-tunnel</ipsec-mode>
<             <enc-algo>aes-128-cbc</enc-algo>
<             <hmac-algo>sha1</hmac-algo>
<             <c2s-spi>0x22BC9F12</c2s-spi>
<             <s2c-spi>0x90505449</s2c-spi>
<             <akey-s2c>
<                 <bits>160</bits>
<                 <val>SOMEKEY1</val>
<             </akey-s2c> 
<             <ekey-s2c>
<                 <bits>128</bits>
<                 <val>SOMEKEY2</val>
<             </ekey-s2c> 
<             <akey-c2s>
<                 <bits>160</bits>
<                 <val>SOMEKEY3</val>
<             </akey-c2s> 
<             <ekey-c2s>
<                 <bits>128</bits>
<                 <val>SOMEKEY4</val>
<             </ekey-c2s> 
<         </ipsec> 
<     </response>
Tunnel timeout (rekey interval) is 180 minutes.
Idle timeout is 180 minutes.
Segmentation fault (core dumped)

```

**Comment by dlenski** [comment]
Thanks, that was exactly what I needed.

Derp derp derp :man_facepalming: … I didn't consider the possibility that `<dns-suffix/>` could be empty. Fixed in 4621a08

**Comment by chessai** [comment]
thank you for the quick fix! works perfectly now.
